azure/Kqlmagic/adal_token_cache.py

```diff
@@ -117,14 +169,107 @@ def _query_cache(self, is_mrrt, user_id, client_id):
                 matches.append(v)
         return matches
 
+    def clear_db(self):
+
+        items = self.db.items()
+
+        for token_tuple in items:
+
+            db_key = token_tuple[0] # 'kqlmagicstore/tokens/USER_ID' address
+            user_id = db_key[len("kqlmagicstore/tokens/"):]
+
+            data = token_tuple[1] #encrypted token
+            if not data:
+                return
+            try:
+                time_created, _unused = Fernet._get_unverified_token_data(data)  #get token timestamp 
+            except InvalidToken:
+                Display.showWarningMessage("Warning: found an illegal token, deleting the token")
+                del self.db["kqlmagicstore/tokens/" + user_id ]
+            EXP_TIME = self.token_exp_time  
+            if(time_created +EXP_TIME < int(time.time()) ):
+                del self.db["kqlmagicstore/tokens/" + user_id ]
+
     def _save(self):
         data = self.serialize()
-        self.db['kqlmagicstore/tokens'] = data
+        data_as_bytes = data.encode()
+
+        #init fernet
+        fernet = self.fernet
+
+        #encrypt 
+        data_encrypted = fernet.encrypt(data_as_bytes)
+        
+        self.db['kqlmagicstore/tokens/'+ str(self.user_id)] = data_encrypted
```

Comment:
this string: 'kqlmagicstore/tokens/' used in many places,  should be derived from Constants.MAGIC_CLASS_NAME.lower()